File: pythonVersion/src/multiform_validator/getOnlyEmail.py
import re

clean_after_default_domain = ['.br', '.io', '.pt', '.us', '.org', '.com']

def getOnlyEmail(text, multiple=False, clean_domain=False, repeat_email=False):
    email_pattern = r'[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}'
    matches = re.findall(email_pattern, text)

    if not matches:
        return 'No email found'

    if clean_domain:
        domains_to_clean = []
        if isinstance(clean_domain, list):
            domains_to_clean = clean_domain
        elif clean_domain is True:
            domains_to_clean = clean_after_default_domain

        cleaned_emails = []
        cleaned_emails2 = []
        for email in matches:
            for domain in domains_to_clean:
                index = email.rfind(domain)
                if index != -1:
                    cleaned_emails.append(email[:index + len(domain)])
                index = email.find(domain)
                if index != -1:
                    cleaned_emails2.append(email[:index + len(domain)])

        if not repeat_email:
            unique_emails = list(set(cleaned_emails2))
            return unique_emails if multiple else unique_emails[0]

        return cleaned_emails2 if multiple else cleaned_emails2[0]

    if not repeat_email:
        unique_emails = list(set(matches))
        return unique_emails if multiple else unique_emails[0]

    return matches if multiple else matches[0]

File: Backup/src/getOnlyEmail.ts

const CleanAfterDefaultDomain: string[] = ['.br', '.io', '.pt', '.us', '.org', '.com'];

/**
 * @example getOnlyEmail("Entre em contato com a equipe:</br> joao@empresa.com, maria@empresa.com, contato@empresa.com", true);
 * // Output: ["joao@empresa.com", "maria@empresa.com", "contato@empresa.com"]
 *
 * @description This function extracts emails from a given text. If multiple is set to true,
 * it returns an array with all emails found. Otherwise, it returns only the first email found as a string.
 *
 * @description Default domainsList is ['.br', '.io', '.pt', '.us', '.org', '.com', ]; you can set true to use this or pass your own list,
 * OBS: Your list gonna replace the default
 * IMPORTANT: List order matters
 *
 * The third parameter solves the following, if the email is formatted as follows, foo@bar.comAAAA, it will clean the email returning only foo@bar.com, all characters that come after the one in the list will be removed and the email comes out clean
 *
 * if you need to skip some param use null
 *
 * @description the fourth parameter, if true, will allow the list to return repeated emails, otherwise the default does not return repeated emails
 *
 * Follow doc: https://tinyurl.com/getOnlyEmail
 *
 * @returns An email string if multiple is false, or an array of email strings if multiple is true.
 */
function getOnlyEmail(text: string, multiple: boolean = false, cleanDomain: boolean|string[] = false, repeatEmail: boolean = false): string|string[] {

  const emailPattern: RegExp = /[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}/g;

  const matches: string|string[]|null = text.match(emailPattern);

  if (!matches) return 'No email found';

  if (cleanDomain) {
		let domainsToClean: string[]; // Certifica-se de que domainsToClean seja sempre um array de strings

		if (Array.isArray(cleanDomain)) {
			domainsToClean = cleanDomain;
		} else if (cleanDomain === true) {
			domainsToClean = CleanAfterDefaultDomain;
		} else {
			domainsToClean = []; // Define como um array vazio se cleanDomain for false ou não for especificado
		}

		const cleanedEmails: string[] = matches.map((email) => {
			for (const domain of domainsToClean) {
				const index: number = email.lastIndexOf(domain);
				if (index !== -1) {
					email = email.substring(0, index + domain.length);
					break; // Break the loop once a match is found
				}
			}

			for (const domain of domainsToClean) {
				const index: number = email.indexOf(domain);
				if (index !== -1) {
					email = email.substring(0, index + domain.length);
					break; // Break the loop once a match is found
				}
			}

			return email;
		});


    if (!repeatEmail) {
      const uniqueEmails: string[] = [...new Set(cleanedEmails)];
      return multiple ? uniqueEmails : uniqueEmails[0];
    }

    return multiple ? cleanedEmails : cleanedEmails[0];
  }

  if (!repeatEmail) {
    const uniqueEmails: string[] = [...new Set(matches)];
    return multiple ? uniqueEmails : uniqueEmails[0];
  }

  return multiple ? matches : matches[0];
	/**
	 * 		const cleanedEmails: string[] = matches.map((email) => {
      for (const domain of domainsToClean) {
        const index: number = email.lastIndexOf(domain);
        if (index !== -1) {
          return email.substring(0, index + domain.length);
        }
      }
      return email;
    });

		const cleanedEmails2: string[] = cleanedEmails.map((email) => {
      for (const domain of domainsToClean) {
        const index: number = email.indexOf(domain);
        if (index !== -1) {
          return email.substring(0, index + domain.length);
        }
      }
      return email;
		});
	 */
}

export default getOnlyEmail;
